src/controllers/logUserPermissionController.js

```diff
@@ -0,0 +1,82 @@
+const moment = require('moment-timezone');
+const UserProfile = require('../models/userProfile');
+const UserPermissionChangeLog = require('../models/userPermissionChangeLog');
+const PermissionChangeLog = require('../models/permissionChangeLog');
+
+const logUserPermissionController = function () {
+  const logPermissionChanges = async function (req, res) {
+    try {
+      const {
+        actualUserProfile,
+        authUser,
+        userId,
+        existingPermissions,
+        addedPermissions,
+        removedPermissions,
+      } = req.body;
+
+      const dateTime = moment().tz('America/Los_Angeles').format();
+
+      const logEntry = new UserPermissionChangeLog({
+        logDateTime: dateTime,
+        userId,
+        individualName: `${actualUserProfile.firstName} ${actualUserProfile.lastName}`,
+        permissions: existingPermissions,
+        permissionsAdded: addedPermissions,
+        permissionsRemoved: removedPermissions,
+        requestorRole: authUser.role,
+        requestorEmail: authUser.email,
+      });
+
+      await logEntry.save();
+      res.status(200).json({ message: 'Permission changes logged successfully' });
+    } catch (error) {
+      console.error('Error logging permission change:', error);
+      res.status(500).json({ error: 'Failed to log permission change' });
+    }
+  };
+
+  const getPermissionChangeLogs = async function (req, res) {
+    try {
+      const userProfile = await UserProfile.findOne({ _id: req.params.userId }).exec();
+
+      if (userProfile) {
+        if (userProfile.role !== 'Owner') {
+          res.status(204).send([]);
```

Comment:
Is this checking for permission to view the history? I think Admins might also want to see the history? not sure.
Could use ``await hasPermission(req.body.requestor, 'getPermissionHistory')`` or something (could use an existing permission or create a new one). There shouldn't be hardcoded references to specific roles if avoidable.

Reply:
I see that's just copied from `getPermissionChangeLogs`... I guess it is only Owners then. Would've been easier to find if this was modifying that file instead of creating a new file.